llvm/include/llvm/IR/IntrinsicsNVVM.td

```diff
@@ -5033,4 +5033,8 @@ def int_nvvm_cp_async_bulk_shared_cta_to_global
        NoCapture<ArgIndex<0>>, NoCapture<ArgIndex<1>>,
        ImmArg<ArgIndex<4>>]>;
 
+
+def int_nvvm_griddepcontrol_launch_dependents: DefaultAttrsIntrinsic<[], []>;
+def int_nvvm_griddepcontrol_wait: DefaultAttrsIntrinsic<[], []>;
```

Comment:
Default attributes *may* be a bit too conservative here. IntrNoMem + IntrHasSideEffects may be a closer match to what we want here. I don't know enough what those instructions do.

If there's doubt, going with the defaults is fine.

Reply:
Thanks for the suggestion! I have updated the intrinsic attributes to `[IntrNoMem, IntrHasSideEffects]` instead of the default ones. Essentially, griddepcontrol acts as a code barrier